src/C_TO_LOGIC.py

```diff
@@ -10,22 +10,26 @@
 import subprocess
 import signal
 import re
+
+from pycparser import c_parser, c_ast, c_generator
+from shutil import which
 from subprocess import Popen, PIPE
 from collections import OrderedDict
 
-def GET_TOOL_PATH(tool_exe_name):
-  from shutil import which
-  w = which(tool_exe_name)
-  if w is not None:
-    return str(w)
-  return None
-
 import VHDL
 import SW_LIB
 import SYN
 import SW_LIB
 import C_TO_FSM
 
+def GET_TOOL_PATH(tool_exe_name):
```

Comment:
I recall getting some funny dependency loop errors or something from python last time I was touching this function (why it seems so awkardly placed)

I dont suppose you saw anything like that?

I can give it a test some time

Reply:
```
julian@home:/media/1TB/Dropbox/PipelineC/git/PipelineC$ ./src/pipelinec 
Traceback (most recent call last):
  File "./src/pipelinec", line 9, in <module>
    import C_TO_LOGIC
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/C_TO_LOGIC.py", line 17, in <module>
    import VHDL
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/VHDL.py", line 10, in <module>
    import SW_LIB
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/SW_LIB.py", line 10, in <module>
    import SYN
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/SYN.py", line 21, in <module>
    import MODELSIM
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/MODELSIM.py", line 10, in <module>
    import SIM
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/SIM.py", line 8, in <module>
    import VERILATOR
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/VERILATOR.py", line 9, in <module>
    import OPEN_TOOLS
  File "/media/1TB/Dropbox/PipelineC/git/PipelineC/src/OPEN_TOOLS.py", line 30, in <module>
    YOSYS_EXE_PATH = C_TO_LOGIC.GET_TOOL_PATH(YOSYS_EXE)
AttributeError: partially initialized module 'C_TO_LOGIC' has no attribute 'GET_TOOL_PATH' (most likely due to a circular import)
```

Reply:
And I didnt have time to find + resolve the circular reference

Perhaps a good idea to do now or avoid...